test/interval_tests/construction.jl

```diff
@@ -26,24 +26,35 @@ const eeuler = Base.MathConstants.e
     @test Interval(1) == Interval(1.0, 1.0)
     @test size(Interval(1)) == (1,)
     @test Interval(big(1)) == Interval(1.0, 1.0)
-    @test Interval(eeuler) == Interval(1.0*eeuler)
     @test Interval(1//10) == Interval{Rational{Int}}(1//10, 1//10)
     @test Interval(BigInt(1)//10) == Interval{Rational{BigInt}}(1//10, 1//10)
     @test Interval( (1.0, 2.0) ) == Interval(1.0, 2.0)
 
     @test Interval{Rational{Int}}(1) == Interval(1//1)
     #@test Interval{Rational{Int}}(pi) == Interval(rationalize(1.0*pi))
 
-    @test Interval{Float64}(pi) == Interval(float(pi))
     @test Interval{BigFloat}(1) == Interval{BigFloat}(big(1.0), big(1.0))
-    @test Interval{BigFloat}(pi) == Interval{BigFloat}(big(pi), big(pi))
 
+    # Irrational
     @test -pi..pi == @interval(-pi,pi)
-    @test 0..pi == hull(interval(0), pi_interval(Float64))
+    @test 0..pi == hull(interval(0), Interval{Float64}(π))
     @test 1.2..pi == @interval(1.2, pi)
-    @test pi..big(4) == hull(pi_interval(BigFloat), interval(4))
-    @test pi..pi == pi_interval(Float64)
-    @test eeuler..pi == hull(@interval(eeuler), pi_interval(Float64))
+    @test pi..big(4) == hull(Interval{BigFloat}(π), interval(4))
+    @test pi..pi == Interval{Float64}(π)
+    @test eeuler..pi == hull(@interval(eeuler), Interval{Float64}(π))
+    @test Interval(π) == @interval(π)
+    @test Interval(π, π) == Interval(π)
+    @test Interval{Float32}(π, π) == Interval{Float32}(π)
+
```

Comment:
I think it is worth adding similar tests (as the three precious lines) for `eeuler` here, since `typeof(eeuler)` returns `Irrational{:ℯ}`.